src/provider.rs

```diff
@@ -0,0 +1,87 @@
+use crate::dc_tools::EmailAddress;
+
+#[derive(Debug, Copy, Clone, PartialEq, ToPrimitive)]
+#[repr(u8)]
+pub enum Status {
+    OK = 1,
+    PREPARATION = 2,
+    BROKEN = 3,
+}
+
+#[derive(Debug)]
+#[repr(u8)]
+pub enum ServerType {
+    SMTP = 1,
+    IMAP = 2,
+}
+
+#[derive(Debug)]
+#[repr(u8)]
+pub enum ServerSocket {
+    STARTTLS = 1,
+    SSL = 2,
+}
+
+#[derive(Debug)]
+pub struct Server {
+    pub stype: ServerType,
+    pub socket: ServerSocket,
+    pub port: u16,
+    pub server: &'static str,
+    pub username: &'static str,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Provider {
+    pub domains: &'static str,
+    pub status: Status,
+    pub before_login_hint: &'static str,
```

Comment:
Can we use an Option<&'static str> instead? this would make for a more friendly api in rust.
Also what about after_login_hint?

Reply:
well, the ui or whatever instance will check for an empty string anyway, there _will_ be errors in the database. also the api does never return NULL, the empty strings says that there is no hint (in general, in the FFI, we avoid NULL where possible).

but as all relate things are in the core, we can easily change this if it become actually an issue.